internal/pkg/template/diff/formatter.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type formatter interface {
+	formateYAML(*yaml.Node) ([]byte, error)
+	formatMod(node diffNode) string
+	formatPath(node diffNode) string
+}
+
+type seqItemFormatter struct{}
+
+func (f *seqItemFormatter) formateYAML(node *yaml.Node) ([]byte, error) {
+	wrapped := &yaml.Node{
+		Kind:    yaml.SequenceNode,
+		Tag:     "!!seq",
+		Content: []*yaml.Node{node},
+	}
+	return yaml.Marshal(wrapped)
+}
+
+func (f *seqItemFormatter) formatMod(node diffNode) string {
+	return fmt.Sprintf("- %s -> %s", node.oldValue().Value, node.newValue().Value)
+}
+
+func (f *seqItemFormatter) formatPath(_ diffNode) string {
+	return ""
+}
+
+type keyedFormatter struct {
+	key string
+}
+
+func (f *keyedFormatter) formateYAML(node *yaml.Node) ([]byte, error) {
+	wrapped := &yaml.Node{
+		Kind: yaml.MappingNode,
+		Tag:  "!!map",
+		Content: []*yaml.Node{
+			{
+				Kind:  yaml.ScalarNode,
+				Tag:   "!!str",
+				Value: f.key,
+			},
+			node,
+		},
+	}
+	return yaml.Marshal(wrapped)
+}
+
+func (f *keyedFormatter) formatMod(node diffNode) string {
+	return fmt.Sprintf("%s: %s -> %s", node.key(), node.oldValue().Value, node.newValue().Value)
```

Comment:
nit: This makes me think that maybe we should rename the methods to this? 💭 
```suggestion
	return fmt.Sprintf("%s: %s -> %s", node.key(), node.oldYAML().Value, node.newYAML().Value)
```